packages/providers/test/provider.test.ts

```diff
@@ -939,6 +931,82 @@ describe('Provider', () => {
     expect(estimateTxSpy).toHaveBeenCalled();
   });
 
+  it('An invalid subscription request throws a FuelError and does not hold the test runner (closes all handles)', async () => {
+    const provider = await Provider.create(FUEL_NETWORK_URL);
+
+    await expectToThrowFuelError(
+      async () => {
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        for await (const value of provider.operations.statusChange({
+          transactionId: 'invalid transaction id',
+        })) {
+          //
+        }
+      },
+
+      { code: FuelError.CODES.INVALID_REQUEST }
+    );
+
+    const response = new TransactionResponse('invalid transaction id', provider);
+
+    await expectToThrowFuelError(() => response.waitForResult(), {
+      code: FuelError.CODES.INVALID_REQUEST,
+    });
+  });
+
+  it('default timeout is undefined', async () => {
+    const provider = await Provider.create(FUEL_NETWORK_URL);
+    expect(provider.options.timeout).toBeUndefined();
+  });
+
+  it('throws TimeoutError on timeout when calling an operation', async () => {
+    const timeout = 500;
+    const provider = await Provider.create(FUEL_NETWORK_URL, { timeout });
+    jest
+      .spyOn(global, 'fetch')
+      .mockImplementationOnce((...args: unknown[]) =>
+        sleep(timeout).then(() =>
+          fetch(args[0] as RequestInfo | URL, args[1] as RequestInit | undefined)
+        )
+      );
+
+    const { error } = await safeExec(async () => {
+      await provider.getBlocks({});
+    });
+
+    expect(error).toMatchObject({
+      code: 23,
+      name: 'TimeoutError',
+      message: 'The operation was aborted due to timeout',
+    });
+  });
+
+  it('throws TimeoutError on timeout when calling a subscription', async () => {
+    const timeout = 500;
+    const provider = await Provider.create(FUEL_NETWORK_URL, { timeout });
+
+    jest
+      .spyOn(global, 'fetch')
+      .mockImplementationOnce((...args: unknown[]) =>
+        sleep(timeout).then(() =>
+          fetch(args[0] as RequestInfo | URL, args[1] as RequestInit | undefined)
+        )
+      );
+
+    const { error } = await safeExec(async () => {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      for await (const iterator of provider.operations.statusChange({
+        transactionId: 'doesnt matter, will be aborted',
+      })) {
+        // shouldn't be reached
```

Comment:
Better to use this than to disable the rule.

```suggestion
        expect(iterator).toBeTruthy(); // shouldn't be reached
```